query/expression.go

```diff
@@ -34,7 +34,8 @@ type EvaluationContext struct {
 	API          api.API          // Api to obtain metadata from
 	Timerange    api.Timerange    // Timerange to fetch data from
 	SampleMethod api.SampleMethod // SampleMethod to use when up/downsampling to match the requested resolution
-	Predicate    api.Predicate
+	Predicate    api.Predicate    // Predicate to apply to TagSets prior to fetching
+	FetchLimit   *int             // A limit on the number of fetches which may be performed
```

Comment:
I think it might be more idiomatic to ship around `*EvaluationContext`. Feels weird to be propagating updates with a pointer in a something passed by value... @jeeyoungk ?